nni/retiarii/experiment/pytorch.py

```diff
@@ -263,37 +193,31 @@ def __init__(self, base_model: nn.Module, evaluator: Union[BaseOneShotTrainer, E
         if evaluator is None:
             raise ValueError('Evaluator should not be none.')
 
-        # TODO: The current design of init interface of Retiarii experiment needs to be reviewed.
+        self.id: str = management.generate_experiment_id()
```

Comment:
This fields are derived. Use `super`.

Reply:
how? the super().__init__() also has other logic that should not be included in RetiariiExperiment. Maybe leave it in the refactor of HPO experiment

Reply:
I don't see anything HPO specific if the argument is a config object or None.

```python
    def __init__(self, config_or_platform: ExperimentConfig | str | list[str] | None):
        nni.runtime.log.init_logger_for_command_line()

        self.config: ExperimentConfig | None = None
        self.id: str = management.generate_experiment_id()
        self.port: int | None = None
        self._proc: Popen | psutil.Process | None = None
        self._action: Literal['create', 'resume', 'view'] = 'create'
        self.url_prefix: str | None = None

        if isinstance(config_or_platform, (str, list)):
            self.config = ExperimentConfig(config_or_platform)
        else:
            self.config = config_or_platform
```